### DISCUSSION TITLE
Derive current version from git tags.

### DISCUSSION THREAD

**Discussion by jpavlav** [callowayproject/bump-my-version#211] [Q&A] [answered]
Hello!

First off, just want to say, thanks for forking the project and keeping it going. I've used the previous incarnations and 👍. Previously, I've used this tool only for `python` projects. Right now, I'm trying to use this tool to create a github action that is somewhat generic for bumping versions.

I thought this would be a cinch, since there is support for detecting the version from a git tag. It would prevent us from having our users include a configuration file in their repo with the version in it. However, I find that it, in the absence of the `current_version` in a config file, the tool bottoms out despite being able to detect the tagged version. Am I just missing some option here? Or is this intentional? 

Examples:

```sh
bump-my-version bump --current-version 0.4.0 --dry-run minor
KeyError: "Key 'current_version' not found at 'tool.bumpversion'"
```

Does `current_version` really need to be present in the config file if the user is specifying the version? Further, does it really need to be present if there is a tagged version available? I see that it detects that:

```sh
bump-my-version bump --current-version 1.0.0 --dry-run
  Specified version (1.0.0) does not match last tagged version (0.4.0)
```

I just wanted to ask about this before I raised an issue. I'd be unopposed to making a contribution to make this possible, but I want to make sure that I'm not stepping outside of the bounds of what is intended. Thanks!

**Comment by jpavlav** [comment]
Just wanted to follow up because I realized something. It does let me show the current version when it is a tag and not in the configuration file:

```
bump-my-version show current_version
0.4.0
```

**Comment by jpavlav** [comment]
I think I kind of just answered my question. This works whether a version is specified in the config file or not:

```
bump-my-version bump --current-version $(git describe --tags --abbrev=0 | cut -d v -f 2) minor --dry-run -vvv
```

**Reply by jpavlav** [reply]
Heh, it doesn't actually work. I forgot I commented out the `update_config_file` function in the `bump` function. That's why it worked.

**Reply by davetapley** [reply]
@coordt is `$(git describe --tags --abbrev=0 | cut -d v -f 2)` really required?

I can't find anything explicit in the docs, but things like:
https://github.com/callowayproject/bump-my-version/blob/5049c88b771bbed7b0da22ad9ba8711b579f9ca6/bumpversion/scm/git.py#L76-L80
and:
https://github.com/callowayproject/bump-my-version/blob/5049c88b771bbed7b0da22ad9ba8711b579f9ca6/bumpversion/scm/git.py#L46-L61

sure make it feel like it should be able to pull the latest tag natively?

This would be nice for me as I build (via PyInstaller) for both Linux and Windows.

**Comment by jpavlav** [comment]
@coordt Do you think it would be outside the spirit of the tool somehow, to allow an option where the config file is not updated? I've tried with both `.bumpversion.toml` and `pyproject.toml`. Including `current_version` in the config file is required even if there is a tag that represents the version.

**Comment by coordt** [comment, marked as answer]
Yes, I believe that should be possible. I think it may be a simple change. I'll look at it this week.

**Reply by jpavlav** [reply]
I went ahead and created an issue from this conversation.
